test/spec/oauthUtil.js

```diff
@@ -601,10 +606,139 @@ describe('validateClaims', function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The issuer [' + claims.iss + '] ' +
+    'does not match [' + validationOptions.issuer + ']'); 
+  });
+
+  it('validates audience', function() {
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: 'nobody'
+    };
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The audience [' + claims.aud + '] ' +
+      'does not match [' + validationOptions.clientId + ']'); 
+  });
+
+  it('validates exp > iat', function() {
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: 1,
+      iat: 2
+    };
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The JWT expired before it was issued'); 
+  });
+
+  it('throws if expired', function() {
+    var now = 10;
+    util.warpToUnixTime(now);
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: now - 1,
+      iat: now - 2
+    };
+    sdk.options.maxClockSkew = 0;
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The JWT expired and is no longer valid'); 
+  });
+
+  it('maxClockSkew extends expiration window', function() {
+    var now = 10;
+    var skew = 2;
+    util.warpToUnixTime(now);
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: now - 1,
+      iat: now - 2
+    };
+    sdk.options.maxClockSkew = skew;
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).not.toThrowError();
+    util.warpToUnixTime(now + skew);
+    expect(fn).toThrowError('The JWT expired and is no longer valid'); 
+  });
+
+  it('throws if issued in the future', function() {
+    var now = 10;
+    util.warpToUnixTime(now);
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: now + 2,
+      iat: now + 1
+    };
+    sdk.options.maxClockSkew = 0;
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The JWT was issued in the future'); 
+  });
+
+  it('maxClockSkew extends iat validation into the future', function() {
+    var now = 10;
+    var skew = 2;
+    util.warpToUnixTime(now);
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: now + 2,
+      iat: now + 1
+    };
+    sdk.options.maxClockSkew = skew;
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).not.toThrowError();
+    util.warpToUnixTime(now - skew);
+    expect(fn).toThrowError('The JWT was issued in the future'); 
```

Comment:
As in the other test, the first case would normally fail. `now` is 10 and `iat` is 11.  So this token was issued in the future. Because of skew we push local time up, 10 + 2 = 12, and therefore the validate does pass. The second test verifies validation still fails if we are too far outside the skew window (now = 8 (+2 skew) = 10 < 11)